Feature/easy install for network

sonique6784 opened the pull request.

sonique6784 commented:
I'm happy to prepare another PR that works together with your work, but don't see why you are blocking this one, it's not introducing any regression and doesn't hurt anyone. If anything it can be merged easily without disturbing your work.

uweseimet commented:
>I suggest to either hold on this PR until yours is merged.

Let's do it as you suggested then, please.

sonique6784 commented:
Thanks. Changes has been tested for wireless (wlan0), but not with wired (eth0)

uweseimet commented:
I see, so this should be tested with eth0 before merging anything.

rdmark commented on easyinstall.sh:
```diff
@@ -291,26 +343,165 @@ function runChoice() {
+
+    # Installing required packages
+    echo "Installing required packages (pytun)"
+    sudo apt-get install -y python3-pip 
+    sudo pip3 install python-pytun
```
I retested my setup steps today, and I believe these packages are actually redundant now.

uweseimet replied:
@rdmark It's interesting that you are pointing this out, because I was already wondering about the relevance of these packages. They are not installed on my Pi, but everything appears to work.

rdmark replied:
@uweseimet yeah good call-out; it was just a vestigial step from an earlier attempt. I set up a new RaSCSI yesterday on a fresh Pi 3B and confirmed that those packages aren't needed.

uweseimet replied:
@rdmark Should I just remove those two lines in uweseimet_develop?

rdmark replied:
@uweseimet Yes I think so

uweseimet replied:
@rdmark I just checked: These lines do not exist in uweseimet_develop. I thought I had seen them in the past, but must have been mistaken.

sonique6784 replied:
Thanks guys I will remove them and retest

rdmark commented on easyinstall.sh:
```diff
@@ -291,26 +343,165 @@ function runChoice() {
+    grep "^denyinterfaces eth0" || sudo echo "denyinterfaces eth0" >> /etc/dhcpcd.conf
+
+    # Setup daynaPORT at start up
+    sudo sed s@"^ExecStart=/usr/local/bin/rascsi"@"ExecStart=/usr/local/bin/rascsi -ID2 daynaport"@ /etc/systemd/system/rascsi.service > /etc/systemd/system/rascsi.service
```
I don't think it's a good idea to do this here, since the host machine may have another SCSI device at ID 2.  One of my Macs that I use with my RaSCSI has an internal hard drive at ID 2 for instance.  I think we should leave it up to the user to configure this.

sonique6784 replied:
Good point. I've added a preliminary check and a warning to the user so they know they should attach DaynaPORT

rdmark replied:
Hm, while it does help certain cases with what you added now, I actually meant another physical SCSI device, not another virtual RaSCSI device. I'm not sure RaSCSI can be aware of the IDs of other physical SCSI device on the chain? 

It just seems inherently risky to me, and something of a noob trap to hard code a parameter into ExecStart in the service config. Beginners will have the facility of the web app UI to manage mounting devices anyways!

BTW, once the bug in https://github.com/akuker/RASCSI/issues/217 gets fixed, I'd suggest we use default.cfg to manage default device setups like this.

sonique6784 replied:
make sense, removed it.

rdmark commented on easyinstall.sh:
```diff
@@ -291,26 +343,165 @@ function runChoice() {
+    sudo apt-get install -y python3-pip 
+    sudo pip3 install python-pytun
+
+    # Configuring bridge
+    grep "rascsi_bridge" /etc/rc.local || echo "
+brctl addbr rascsi_bridge
+ifconfig rascsi_bridge $ROUTER_IP/24 up
+" >> /etc/rc.local
```
As @uweseimet called out elsewhere, the current uweseimet_develop contains code that makes this part redundant.  I'd suggest removing it in anticipation of that code being merged very soon.

sonique6784 replied:
I don't think with the current develop network can work without. I'm going to prepare an updated version that work with uweseimet_develop branch next.

rdmark commented on easyinstall.sh:
```diff
@@ -325,9 +516,10 @@ function showMenu() {
     echo "CREATE EMPTY DRIVE"
     echo "  6) 600MB drive (recommended)"
     echo "  7) custom drive size (up to 4000MB)"
+    echo "NETWORK"
+    echo "  8) Network assistant (wire and wireless)"
```
I think we should give the user the option to pick which one to configure, ethernet or wifi, since the two provide very different capabilities. Ethernet gives you DHCP and access to AppleTalk with other Macs on the network, while wifi sets up an isolated network with static IP.

sonique6784 replied:
at the moment the assistant detect what configuration (wired (eth0) or wireless (wlan0)) Line 449 `checkNetworkType()`
I've modified the code, so if both interfaces are found, the user can choose the one they want to configure.

uweseimet replied:
What about other interfaces, e.g. eth1?

sonique6784 replied:
no support yet. but can be added later. like I said earlier I don't have the hardware for wired configuration.
if we looking at having a 100% perfect solution, then let's close this PR.

uweseimet replied:
This is not only about eth0, but also wlan1, for instance. The idea (as already discussed) is to use the rascsi from the uwe_develop branch, which supports any network interface.
Testing can probably be done by somebody else. I have a USB network adapter, which provides wlan1 for my Pi.

sonique6784 replied:
I've added something to select the interface and configure the bridge properly too.

uweseimet replied:
I think you can actually test your changes yourself by adding more network interfaces to your Pi:

```
>ip link add eth1 type dummy
>ifconfig eth1 200.200.200.200 up
>ifconfig eth1

eth1: flags=195<UP,BROADCAST,RUNNING,NOARP>  mtu 1500
        inet 200.200.200.200  netmask 255.255.255.0  broadcast 200.200.200.255
        inet6 fe80::ea16:8c92:647c:ca44  prefixlen 64  scopeid 0x20<link>
        ether 3e:ab:59:5c:4c:dd  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 17  bytes 2174 (2.1 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

sonique6784 replied:
thanks very much! I will proceed to testing wired and multiple interface

sonique6784 replied:
testing looks good btw

rdmark commented:
@sonique6784 I adapted your solution here to work with uweseimet_develop over at this PR: https://github.com/akuker/RASCSI/pull/229
To get something that works reliably with minimal complexity in the bash script, I opted for a simple solution that I believe works for 99% of use cases: Someone either using eth0 or wlan0, and no other interfaces.
The reason is that eth0 and wlan0 are the defaults that rascsi will fall back to. And if you want to use something else you need to pass parameters to rascsi/rasctl when you attach the DaynaPort device, which I think is beyond the scope of this script.
What do you think?

sonique6784 commented:
Hi @rdmark , my solution has a module to detect interface but also to offer the user to select, feel free to pick it. I think that something @uweseimet wanted to cover all cases

rdmark commented:
@sonique6784 I adopted your solution for detecting network interfaces and added flows to create custom network configurations over at my PR. It needs some cleanup but seems to be working, although I don't have a multi-interface environment to test in.

uweseimet commented:
@rdmark I already commented this somewhere (can't find it right now), and @sonique6784 is also aware of this: With "ifconfig" or "ip link add" you can add additional virtual network interface of any name, i.e. testing does not require physical network interfaces.

rdmark commented:
@uweseimet Good idea. I've tested and improved the script now. I think it should be good to go, but let me know if you can think of any improvements.

sonique6784 commented:
Hi @uweseimet and @rdmark , I did further testing and this is ready for merge, awaiting for your review.

rdmark commented:
@sonique6784 May I suggest we hold off on this PR until Uwe's and mine have been merged to develop?  As I mentioned above, I've made a slightly different solution to the same problem in my PR that better matches Uwe's development code (plus many other additions to the script).  If we merge your change onto develop now, there will be unsolvable merge conflicts later.

sonique6784 commented:
> @sonique6784 May I suggest we hold off on this PR until Uwe's and mine have been merged to develop? As I mentioned above, I've made a slightly different solution to the same problem in my PR that better matches Uwe's development code (plus many other additions to the script). If we merge your change onto develop now, there will be unsolvable merge conflicts later.

the all point of this PR is to have network configuration without having to wait. this PR is no harm. I've addressed all the concerns you guys had.

uweseimet commented:
@sonique6784 This means that your changes are now making use of the new network features, which have been merged into the develop branch? (Don't have the time right now to look this up.)
